app/v2/controllers/CreateHistoricNonFHLUkPiePeriodSummaryController.scala

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ */
+
+package v2.controllers
+
+import cats.data.EitherT
+import cats.implicits._
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, ControllerComponents}
+import utils.{IdGenerator, Logging}
+import v2.controllers.requestParsers.CreateHistoricNonFhlUkPropertyPeriodSummaryRequestParser
+import v2.hateoas.HateoasFactory
+import v2.models.errors._
+import v2.models.request.createHistoricNonFhlUkPropertyPeriodSummary.CreateHistoricNonFhlUkPropertyPeriodSummaryRawData
+import v2.models.response.createHistoricNonFhlUkPiePeriodSummary.CreateHistoricNonFhlUkPiePeriodSummaryHateoasData
+import v2.services.{CreateHistoricNonFhlUkPropertyPeriodSummaryService, EnrolmentsAuthService, MtdIdLookupService}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+/*
+  Pie = Property Income & Expenses
+*/
+@Singleton
+class CreateHistoricNonFHLUkPiePeriodSummaryController @Inject()(val authService: EnrolmentsAuthService,
+                                                                 val lookupService: MtdIdLookupService,
+                                                                 parser: CreateHistoricNonFhlUkPropertyPeriodSummaryRequestParser,
+                                                                 service: CreateHistoricNonFhlUkPropertyPeriodSummaryService,
+                                                                 hateoasFactory: HateoasFactory,
+                                                                 cc: ControllerComponents,
+                                                                 idGenerator: IdGenerator)(implicit ec: ExecutionContext)
+  extends AuthorisedController(cc)
+    with BaseController
+    with Logging {
+
+  implicit val endpointLogContext: EndpointLogContext =
+    EndpointLogContext(controllerName = "CreateAmendHistoricNonFHLUkPiePeriodSummaryController",
+      endpointName = "CreateAmendHistoricNonFHLUkPropertyIncomeExpensesPeriodSummary")
+
+  def handleRequest(nino: String): Action[JsValue] =
+    authorisedAction(nino).async(parse.json) { implicit request =>
+      implicit val correlationId: String = idGenerator.getCorrelationId
+
+      logger.info(message = s"[${endpointLogContext.controllerName}][${endpointLogContext.endpointName}] with correlationId : $correlationId")
+
+      val rawData = CreateHistoricNonFhlUkPropertyPeriodSummaryRawData(nino, request.body)
+      val result =
+        for {
+          parsedRequest   <- EitherT.fromEither[Future](parser.parseRequest(rawData))
+          serviceResponse <- EitherT(service.createPeriodSummary(parsedRequest))
+          vendorResponse <- EitherT.fromEither[Future](
+            hateoasFactory
+              .wrap(serviceResponse.responseData, CreateHistoricNonFhlUkPiePeriodSummaryHateoasData(nino, serviceResponse.responseData.periodId.get, serviceResponse.responseData.transactionReference))
```

Comment:
Best not to use unprotected `get` on an `Option`. Form the periodId from the request body instead.

Reply:
Updated